src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/ModuleTokenResolver.cs

```diff
@@ -436,5 +438,123 @@ public DummyTypeInfo GetTypeFromSpecification(MetadataReader reader, ModuleToken
                 return DummyTypeInfo.Instance;
             }
         }
+
+        private ImmutableHashSet<MethodDesc> KnownMethods;
+        private ImmutableHashSet<TypeDesc> KnownTypes;
+
+        private void InitializeKnownMethodsAndTypes()
+        {
+            var knownTypes = ImmutableHashSet.CreateBuilder<TypeDesc>();
+            var knownMethods = ImmutableHashSet.CreateBuilder<MethodDesc>();
+
+            var exception = CompilerContext.SystemModule.GetKnownType("System"u8, "Exception"u8);
+            knownTypes.Add(exception);
+
+            // AsyncHelpers type and methods
+            var asyncHelpers = CompilerContext.SystemModule.GetKnownType("System.Runtime.CompilerServices"u8, "AsyncHelpers"u8);
+            knownTypes.Add(asyncHelpers);
+
+            knownMethods.Add(asyncHelpers.GetKnownMethod("AsyncCallContinuation"u8, null));
+            knownMethods.Add(asyncHelpers.GetKnownMethod("TaskFromException"u8, new MethodSignature(MethodSignatureFlags.Static, 0, CompilerContext.SystemModule.GetKnownType("System.Threading.Tasks"u8, "Task"u8), new TypeDesc[] { exception })));
+            knownMethods.Add(asyncHelpers.GetKnownMethod("ValueTaskFromException"u8, new MethodSignature(MethodSignatureFlags.Static, 0, CompilerContext.SystemModule.GetKnownType("System.Threading.Tasks"u8, "ValueTask"u8), new TypeDesc[] { exception })));
+            knownMethods.Add(asyncHelpers.GetKnownMethod("TaskFromException"u8, new MethodSignature(MethodSignatureFlags.Static, 1, CompilerContext.SystemModule.GetKnownType("System.Threading.Tasks"u8, "Task`1"u8).MakeInstantiatedType(CompilerContext.GetSignatureVariable(0, method: true)), new TypeDesc[] { exception })));
+            knownMethods.Add(asyncHelpers.GetKnownMethod("ValueTaskFromException"u8, new MethodSignature(MethodSignatureFlags.Static, 1, CompilerContext.SystemModule.GetKnownType("System.Threading.Tasks"u8, "ValueTask`1"u8).MakeInstantiatedType(CompilerContext.GetSignatureVariable(0, method: true)), new TypeDesc[] { exception })));
+            knownMethods.Add(asyncHelpers.GetKnownMethod("FinalizeTaskReturningThunk"u8, new MethodSignature(MethodSignatureFlags.Static, 0, CompilerContext.SystemModule.GetKnownType("System.Threading.Tasks"u8, "Task"u8), Array.Empty<TypeDesc>())));
```

Comment:
Given that this is somewhat setting the names of these methods in stone, we should make sure that they have good names, following the same principles we use to name public APIs.

I would expect that's called `FinalizeFoo` to take `Foo` as an argument. It is not what this method does. This can use a better name.